pkg/virtualmachine/enricher/enricher_impl.go

```diff
@@ -0,0 +1,65 @@
+package enricher
+
+import (
+	"context"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/pkg/errors"
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/scanners/scannerv4"
+	"github.com/stackrox/rox/pkg/scannerv4/client"
+)
+
+var (
+	log         = logging.LoggerForModule()
+	scanTimeout = env.ScanTimeout.DurationSetting()
+)
+
+const vmMockDigest = "vm-registry/repository@sha256:deadb33fdeadb33fdeadb33fdeadb33fdeadb33fdeadb33fdeadb33fdeadb33f"
+
+type enricherImpl struct {
+	scannerClient client.Scanner
+}
+
+func New(scannerClient client.Scanner) VirtualMachineEnricher {
+	return &enricherImpl{
+		scannerClient: scannerClient,
+	}
+}
+
+func (e *enricherImpl) EnrichVirtualMachineWithVulnerabilities(vm *storage.VirtualMachine, indexReport *v4.IndexReport) error {
+	// Clear any pre-existing notes
+	vm.Notes = vm.Notes[:0]
+
+	if e.scannerClient == nil {
+		vm.Notes = append(vm.Notes, storage.VirtualMachine_MISSING_SCAN_DATA)
+		return errors.New("Scanner V4 client not available for VM enrichment")
+	}
+
+	if indexReport == nil {
+		vm.Notes = append(vm.Notes, storage.VirtualMachine_MISSING_SCAN_DATA)
+		return errors.New("index report is required for VM scanning")
+	}
+
+	vmDigest, err := name.NewDigest(vmMockDigest)
```

Comment:
Propagating review comment from #16609 
> nit: might as well do this within the `var` block or in an `init` so we don't have to keep recreating this

Note: the pattern is the same as the one for the node scanner in `pkg/scanners/scannerv4`, but the improvement could be considered.